internal/workflow/preprocessing.go

```diff
@@ -1,25 +1,44 @@
 package workflow
 
 import (
-	"errors"
 	"fmt"
 	"path/filepath"
+	"strings"
 	"time"
 
 	"github.com/artefactual-sdps/temporal-activities/bagit"
 	"go.artefactual.dev/tools/temporal"
+	temporalsdk_log "go.temporal.io/sdk/log"
 	temporalsdk_temporal "go.temporal.io/sdk/temporal"
 	temporalsdk_workflow "go.temporal.io/sdk/workflow"
 
 	"github.com/artefactual-sdps/preprocessing-sfa/internal/activities"
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/enums"
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/eventlog"
+)
+
+type Outcome int
+
+const (
+	OutcomeSuccess Outcome = iota
+	OutcomeSystemError
+	OutcomeContentError
 )
 
 type PreprocessingWorkflowParams struct {
 	RelativePath string
 }
 
 type PreprocessingWorkflowResult struct {
+	Outcome      Outcome
 	RelativePath string
+	Events       []eventlog.Event
```

Comment:
Feel free to ignore, but I'll insist. I think `Events` is too general if they are only meant to create preservation tasks in Enduro, I'd call them `PreservationTasks` directly.